ui/core/individual_sim_ui.tsx

```diff
@@ -692,4 +692,40 @@ export abstract class IndividualSimUI<SpecType extends Spec> extends SimUI {
 			}
 		});
 	}
+
+	// Determines whether this sim has either a hard cap or soft cap configured for a particular PseudoStat. Used by the stat
+	// weights code to ensure that school-specific EPs are calculated for Rating stats whenever school-specific caps are present.
+	hasCapForPseudoStat(pseudoStat: PseudoStat): boolean {
+		// First check both default and currently stored hard caps.
+		const defaultHardCaps = this.individualConfig.defaults.statCaps || new Stats();
+
+		if ((defaultHardCaps.getPseudoStat(pseudoStat) != 0) || (this.player.getStatCaps().getPseudoStat(pseudoStat) != 0)) {
```

Comment:
I still think for conventions sake we should always do `===` or `!==` to check strict equality, unless you want `false` to also match `0`?

Reply:
Sounds good, thanks. It doesn't matter for this example since `getPseudoStat()` always returns a `number`, but will keep this in mind when when writing Typescript code.